modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -18,56 +18,71 @@ import {useSelectModel} from './useSelectModel';
-    const {localRef, elementRef} = useLocalRef<HTMLInputElement>(
-      useForkRef(ref as React.Ref<HTMLElement>, ref)
-    );
+  createElemPropsHook(useSelectModel)(
+    (model, ref, elemProps: {keySofar?: string; placeholder?: string} = {}) => {
+      const {localRef, elementRef} = useLocalRef<HTMLInputElement>(
+        useForkRef(ref as React.Ref<HTMLElement>, ref)
+      );
 
-    useResizeObserver({
-      ref: localRef,
-      onResize: data => {
-        if (model.state.visibility === 'visible') {
-          // Width of the Input + 2px border + 8px padding
-          const calculatedWidth = data.width ? data.width + 42 + 8 : 0;
-          model.events.setWidth(calculatedWidth);
-        }
-      },
-    });
-
-    return {
-      onKeyDown(event: React.KeyboardEvent) {
-        // Prevent the keys from being enter in the input
-        if (event.key !== 'Tab') {
-          event.preventDefault();
-        }
+      useResizeObserver({
+        ref: localRef,
+        onResize: data => {
+          if (model.state.visibility === 'visible') {
+            // Width of the Input + 2px border + 8px padding
+            const calculatedWidth = data.width ? data.width + 42 + 8 : 0;
+            model.events.setWidth(calculatedWidth);
+          }
+        },
+      });
 
-        // Select should open if Spacebar is typed and nothing has been typed AND the menu is hidden
+      React.useEffect(() => {
         if (
-          event.key === 'Spacebar' ||
-          (event.key === ' ' && model.state.visibility === 'hidden' && elemProps?.keySofar === '')
+          model.state.inputRef.current &&
+          model.state.items.length > 0 &&
+          model.state.selectedIds[0]
         ) {
-          model.events.show();
+          model.state.inputRef.current.value = model.navigation.getItem(
+            model.state.selectedIds[0],
+            model
+          ).textValue;
         }
-        // if the menu is visible
-        if (
-          (event.key === 'Spacebar' || event.key === ' ') &&
-          model.state.visibility === 'visible'
-        ) {
-          // If key so far is empty, they're done typing, select the item where the cursor is located and hide the menu
-          if (elemProps?.keySofar === '') {
-            model.events.select({
-              id: model.state.cursorId,
-            });
-            model.events.hide();
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, [model.state.inputRef, model.state.items]);
+
+      return {
+        onKeyDown(event: React.KeyboardEvent) {
+          // Prevent the keys from being enter in the input
+          if (event.key !== 'Tab') {
+            event.preventDefault();
           }
-        }
-      },
-      ref: elementRef,
-      autoComplete: 'off',
-      defaultValue:
-        model.navigation.getItem(model.state.selectedIds[0], model).textValue || model.state.value,
```

Comment:
Should this be removed? I understand that an error should not be thrown by `model.navigation.getItem` if no items are loaded yet, but it seems like a regression to completely remove a `defaultValue` for all use-cases.